src/core/reactor.cc

```diff
@@ -2218,6 +2219,41 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<std::optional<struct group>> reactor::getgrnam(std::string_view name, char *buf, size_t buflen) {
+    return futurize_invoke([name, buf, buflen, this] {
+        return _thread_pool->submit<syscall_result_extra<std::optional<struct group>>>([name = sstring(name), buf, buflen] {
+            errno = 0;
+            struct group grp;
+            struct group *result;
+            memset(&grp, 0, sizeof(struct group));
+            int ret = ::getgrnam_r(name.c_str(), &grp, buf, buflen, &result);
+            if (!result) {
```

Comment:
quote from `getgrnam(3)`

> On  success, getgrnam_r() and getgrgid_r() return zero, and set *result to grp.  If no matching group record was found, these functions return 0 and store NULL in *result.

so i'd suggest check `ret` instead of `result` here.